src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -227,6 +226,40 @@ void RunCKSolution(const Handle& handle,
     }
 }
 
+InvokerFactory MakeInvokerFactoryHipImplGemmFwdXdlops(const ProblemDescription& problem, 
+    const PerformanceConfigHipImplicitGemmFwdXdlops& config) {
+
+  auto ck_args = CKArgs{problem};
+  miopenDataType_t data_type = problem.conv_problem.GetInDataType();
+  auto config_idx = config.index;
+
+
+  InvokerFactory ret = [ck_args, data_type, config_idx] (const std::vector<Kernel>& kernels) {
```

Comment:
```suggestion
  InvokerFactory ret = [ck_args = std::move(ck_args), data_type, config_idx] (const std::vector<Kernel>& kernels) {
```

Reply:
Alternatively, the factory can be implemented as a class/structure with constructor and `operator ()`. We don't really use that pattern anywhere, but `std::function` should accept any functor with a valid signature. That is kind of what lambdas compile to anyway.

Reply:
Good catch on the `move` optimization.

Reply:
added `move` capture in 23d8931

Reply:
Happy to have a discussion about how to improve `InvokerFactory` pattern but IMHO it's beyond the scope of this PR.

Reply:
@amberhassaan It's sad, but this is exactly what the 2nd bullet of #2301 is about ;)

Reply:
@amberhassaan Oops, how I see what you mean. Well, what @DrizztDoUrden is trying to tell, is that InvokerFactory can be implemented as "a class/structure with constructor and operator()" -- which may be thought as just another, more explicit way of implementing a lambda. The Invoker/Factory machinery and its use patterns remain the same.